### PR TITLE
Implemented getCurrentRouteIndex and getCurrentRoute to Navigator

### PR DESCRIPTION
**Pull request by Sunakujira1**
I had issue figuring out which route I was at, even though `getCurrentRoutes()` exists.
These two functions should cover the use cases when dealing with the route stack.

### CODE REVIEW

File: Libraries/CustomComponents/Navigator/Navigator.js [hunk 1]
```diff
@@ -1043,6 +1043,14 @@ var Navigator = React.createClass({
     return this.state.routeStack.slice();
   },
 
+  getCurrentRouteIndex: function() {
+    return this.state.presentedIndex;
+  },
```
Comment (grabbou) [inline Libraries/CustomComponents/Navigator/Navigator.js]: Is this function needed if current route can be retrieved separately?

File: Libraries/CustomComponents/Navigator/Navigator.js [hunk 2]
```diff
@@ -1043,6 +1043,14 @@ var Navigator = React.createClass({
...
+
+  getCurrentRoute: function() {
+    return {...this.state.routeStack[this.getCurrentRouteIndex()]};
```
Comment (grabbou) [inline Libraries/CustomComponents/Navigator/Navigator.js]: Do we have to make a copy of a scene here? When you get an array of routes, we just make sure you do not mutate stack manually (however underlaying scenes are still the same).

**Comment by Sunakujira1** [comment]
@grabbou Thank you for having a look!

> Is this function needed if current route can be retrieved separately?

I can see this being useful when handling different case at specific index. For example, exit the program using BackAndroid when there is no scene to go back to (index 0).

> Do we have to make a copy of a scene here? When you get an array of routes, we just make sure you do not mutate stack manually (however underlaying scenes are still the same).

I believe we do not want to mutate the particular route as well (Should use `replace(route)` in that case), thus I mimicked the shallow copy strategy of `getCurrentStates()` (Although both are just a shallow copy).

File: Libraries/CustomComponents/Navigator/Navigator.js (hunk 2 above)
Comment (hedgerwang) [inline Libraries/CustomComponents/Navigator/Navigator.js]: I really hope that we could stop adding new getters to expose the internal state of the navigator. Once we move toward the new controlled-view model, it won't make much sense to let the component hold the state.

Comment (Sunakujira1) [inline Libraries/CustomComponents/Navigator/Navigator.js]: @hedgerwang I am not familiar with how the navigator will look in future, but for current, I needed this feature for my certain use case. I see some other questionable things about navigator, and I hope they get fixed in future as well.

**Comment by facebook-github-bot-5** [comment]
@ericvicenti would you mind taking a look at this pull request? It's been a while since the last commit was reviewed.

**Comment by mkonicek** [comment]
@Sunakujira1 What use case do you need `getCurrentRoute` and `getCurrentRouteIndex` for? Trying to better understand this pull request.

**Comment by lacker** [comment]
I think this pull request has been abandoned, so I am going to close it. Please feel free to reopen if you do intend to work on this further!

**Pull request closed without merging**